src/config.cpp

```diff
@@ -53,6 +53,11 @@ struct adl_serializer<nonstd::optional<T>> {
 namespace bbp {
 namespace sonata {
 
+NLOHMANN_JSON_SERIALIZE_ENUM(CircuitConfig::Type,
+                             {{CircuitConfig::Type::invalid, nullptr},
+                              {CircuitConfig::Type::Partial, "partial"},
+                              {CircuitConfig::Type::Complete, "complete"}})
```

Comment:
It might be worth checking if the capitalization is consistent. We have `Type::invalid` lower case `i`, then `Type::Partial` but `SpikeLocation::soma`. The string identifiers are all lower case.

Reply:
yeah, I didn't like the difference in case either, but I was following what had been done before.

I need to check if `NLOHMANN_JSON_SERIALIZE_ENUM` is case sensitive; sadly the spec has mixed case, so we may be stuck :(

Reply:
I can confirm that `NLOHMANN_JSON_SERIALIZE_ENUM` is case sensitive...